### DISCUSSION TITLE
Token to request ChatGPT thru a proxy

### DISCUSSION THREAD

**Discussion by ThierryBachmann** [RageAgainstThePixel/com.openai.unity#183] [Q&A] [answered]
A request to ChatGPT thru the proxy discord.pawan doesn't work.

the token prefix` sess-` is retained in the bearer.

To check, a brutal action by removing the 5 first characters has corrected the issue.
```csharp
headers.Add("Authorization", Rest.GetBearerOAuthToken(Authentication.Info.ApiKey.Substring(5)));
```
Perhaps, there is something more cleverly to correct.

**Comment by StephenHodgson** [comment]
I designed the proxy service to reflect a session token based mechanism similar to the one openAI uses for chat gpt.

More info here https://github.com/RageAgainstThePixel/OpenAI-DotNet/blob/main/OpenAI-DotNet-Proxy/Readme.md

If this is a proxy service you control, I'd prefer you use the `sess-` prefix. If it is not, then maybe contact the maintainer and request that they support it. This is probably not something I will be changing.

**Reply by StephenHodgson** [reply, marked as answer]
@ThierryBachmann Checkout the latest release. #185

**Comment by ThierryBachmann** [comment]
it's a good practice, sk- for chatGpt, pk- for pawan. 
But, why adding another prefix before the prefix? 
Why create a dependency between your API and pawan?
Clearly, no one will be able to use your framework with pawan or any others proxies.

**Reply by StephenHodgson** [reply]
I don't mind adding the ability to set a different prefix. This should be possible I think.

> Clearly, no one will be able to use your framework with pawan or any others proxies.

I built this specifically for people managing their own proxies, not necessarily using a 3rd party one.
Personally, I would be apprehensive to do so for security concerns.

**Comment by ThierryBachmann** [comment]
In OpenAIClient.cs we have this check:

    (!Authentication.Info.ApiKey.Contains(OpenAIAuthInfo.SecretKeyPrefix) &&
     !Authentication.Info.ApiKey.Contains(OpenAIAuthInfo.SessionKeyPrefix))))

So only key starting with sess- or sk- are authorized.
One possibilities is to open this choices with your API to others prefixes (pk- for pawan proxy)

**Reply by StephenHodgson** [reply]
Sounds good. I think being able to customize the prefix should be possible.

**Comment by ThierryBachmann** [comment]
Finally, Pawan proxy seems accept prefix like sess-pk- 
I got another issue from Pawan (created attribut is not in second and not suit in an integer)
Obviously, I will check with Pawan for this issue.
thanks
